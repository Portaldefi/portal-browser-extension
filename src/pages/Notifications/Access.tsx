import React from 'react';
import { Button, Grid, Header, 
  // Icon, 
  List } from 'semantic-ui-react';

import AccessOption from '../../components/AccessOption';

export default () => {

  const closeWindow = () => {
    window.close();
  }

  return (
    <div className='mainframe'>
      <Grid className='bg-white py-12'>
        <Grid.Row centered>
          <Header as='h1' className='title'>Grove Chat</Header>
          <p className='font-size-14 font-weight-400 line-height-17 font-family-Inter'>is requesting access to your Fabric Identity</p>
        </Grid.Row>
        <Grid.Row stretched centered>
          <div className='light-shadow p-16 py-9 w-100 font-family-Inter font-weight-400 bg-white'>
            <p className='font-size-12 line-height-15 text-align-left mb-3'>Website</p>
            <p className='font-size-14 line-height-17 text-align-left'>https://grove.chat/</p>
          </div>
          <div className='light-shadow p-16 py-9 w-100 font-family-Inter font-weight-400 bg-white'>
            <p className='font-size-12 line-height-15 text-align-left mb-3'>Fabric Identity</p>
            <p className='font-size-14 line-height-17 text-align-left'>fabricm3fuk238hJ3oawHIlk3wa4kjls</p>
          </div>
        </Grid.Row>
        <Grid.Row className='p-none'>
          <p className='font-family-Inter font-weight-400 font-size-12 line-height-15 px-16 m-none'>Allow access for</p>
          <List className='w-100'>
            <List.Item className='p-none'>
              <List.Content>
                <AccessOption name='This session' comment={''} />
              </List.Content>
            </List.Item>
            <List.Item className='p-none'>
              <List.Content>
                <AccessOption name='Indefinitely' comment={''} />
              </List.Content>
            </List.Item>
          </List>
        </Grid.Row>
        <Grid.Row centered columns={2}>
          <Grid.Column className='px-16'>
            <Button className='secondary-button' onClick={closeWindow}>Reject</Button>
          </Grid.Column>
          <Grid.Column className='px-16'>
            <Button className='primary-button' onClick={closeWindow}>Allow</Button>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </div>
  );
}