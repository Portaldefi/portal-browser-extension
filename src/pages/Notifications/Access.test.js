import Access from './Access';
import { create, act } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Button } from 'semantic-ui-react';

describe('AccessComponent', () => {
  it('should work', () => {
    let tree;
    act(() => {
      tree = create(
        <BrowserRouter>
          <Access />
        </BrowserRouter>
      );
    });

    expect(tree).toMatchSnapshot();
  });
});
