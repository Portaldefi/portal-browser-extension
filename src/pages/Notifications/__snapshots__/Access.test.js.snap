// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessComponent should work 1`] = `
<div
  className="mainframe"
>
  <div
    className="ui grid bg-white py-12"
  >
    <div
      className="centered row"
    >
      <h1
        className="ui header title"
      >
        Grove Chat
      </h1>
      <p
        className="font-size-14 font-weight-400 line-height-17 font-family-Inter"
      >
        is requesting access to your Fabric Identity
      </p>
    </div>
    <div
      className="centered stretched row"
    >
      <div
        className="light-shadow p-16 py-9 w-100 font-family-Inter font-weight-400 bg-white"
      >
        <p
          className="font-size-12 line-height-15 text-align-left mb-3"
        >
          Website
        </p>
        <p
          className="font-size-14 line-height-17 text-align-left"
        >
          https://grove.chat/
        </p>
      </div>
      <div
        className="light-shadow p-16 py-9 w-100 font-family-Inter font-weight-400 bg-white"
      >
        <p
          className="font-size-12 line-height-15 text-align-left mb-3"
        >
          Fabric Identity
        </p>
        <p
          className="font-size-14 line-height-17 text-align-left"
        >
          fabricm3fuk238hJ3oawHIlk3wa4kjls
        </p>
      </div>
    </div>
    <div
      className="row p-none"
    >
      <p
        className="font-family-Inter font-weight-400 font-size-12 line-height-15 px-16 m-none"
      >
        Allow access for
      </p>
      <div
        className="ui list w-100"
        role="list"
      >
        <div
          className="item p-none"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="content"
          >
            <div
              className="ui grid p-16 py-5 bg-white"
            >
              <div
                className="left aligned eleven wide column"
              >
                <p
                  className="font-family-Inter font-weight-400 font-size-14 line-height-17"
                >
                  This session
                </p>
              </div>
              <div
                className="middle aligned five wide column"
              >
                <div
                  className="ui fitted radio checkbox"
                  onChange={[Function]}
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                >
                  <input
                    checked={false}
                    className="hidden"
                    readOnly={true}
                    tabIndex={0}
                    type="radio"
                  />
                  <label />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="item p-none"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="content"
          >
            <div
              className="ui grid p-16 py-5 bg-white"
            >
              <div
                className="left aligned eleven wide column"
              >
                <p
                  className="font-family-Inter font-weight-400 font-size-14 line-height-17"
                >
                  Indefinitely
                </p>
              </div>
              <div
                className="middle aligned five wide column"
              >
                <div
                  className="ui fitted radio checkbox"
                  onChange={[Function]}
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                >
                  <input
                    checked={false}
                    className="hidden"
                    readOnly={true}
                    tabIndex={0}
                    type="radio"
                  />
                  <label />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="centered two column row"
    >
      <div
        className="column px-16"
      >
        <button
          className="ui button secondary-button"
          onClick={[Function]}
        >
          Reject
        </button>
      </div>
      <div
        className="column px-16"
      >
        <button
          className="ui button primary-button"
          onClick={[Function]}
        >
          Allow
        </button>
      </div>
    </div>
  </div>
</div>
`;
